File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/SBPacketChunkRequest.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;
import pize.tests.voxelgame.server.net.PlayerGameConnection;

import java.io.IOException;

public class SBPacketChunkRequest extends IPacket<PlayerGameConnection>{
    
    public static final int PACKET_ID = 12;
    
    public SBPacketChunkRequest(){
        super(PACKET_ID);
    }
    
    public int chunkX, chunkZ;
    
    public SBPacketChunkRequest(int chunkX, int chunkZ){
        this();
        
        this.chunkX = chunkX;
        this.chunkZ = chunkZ;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeInt(chunkX);
        stream.writeInt(chunkZ);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        chunkX = stream.readInt();
        chunkZ = stream.readInt();
    }
    
    @Override
    public void handle(PlayerGameConnection packetHandler){
        packetHandler.handleChunkRequest(this);
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/SBPacketChunkRequest.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.tests.voxelgame.server.net.PlayerGameConnection
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class SBPacketChunkRequest() : IPacket<PlayerGameConnection?>(PACKET_ID) {
    var chunkX = 0
    var chunkZ = 0

    constructor(chunkX: Int, chunkZ: Int) : this() {
        this.chunkX = chunkX
        this.chunkZ = chunkZ
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeInt(chunkX)
        stream.writeInt(chunkZ)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        chunkX = stream!!.readInt()
        chunkZ = stream.readInt()
    }

    override fun handle(packetHandler: PlayerGameConnection) {
        packetHandler.handleChunkRequest(this)
    }

    companion object {
        const val PACKET_ID = 12
    }
}
